src/base/static/components/molecules/form-field-types/geolocate-field.js

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "react-emotion";
+
+import { Button } from "../../atoms/buttons";
+import { FontAwesomeIcon } from "../../atoms/imagery";
+
+import emitter from "../../../utils/emitter";
+import constants from "../../../constants";
+
+const GeolocateButton = styled(Button)(props => ({
+  display: "flex",
+  alignItems: "center",
+}));
+
+const GeolocatePlaceholder = styled("span")({
+  paddingLeft: "8px",
+  textTransform: "none",
+  textAlign: "left",
+  color: "#999",
+});
+
+const GeolocateField = props => {
+  return (
+    <GeolocateButton
+      onClick={() => emitter.emit(constants.TRIGGER_GEOLOCATE_EVENT)}
+    >
+      <FontAwesomeIcon content="\f05b" fontSize="1.5rem" />
```

Comment:
nit: I would prefer to save the unicode as constants within our atoms, and pass in a description of the content through the api (eg: `content="globe"`). It'll make the api easier to work with, and keeps the unicode in one place.

Reply:
Would we need a complete mapping of all the FontAwesome icons in that case? An alternative approach would be to use the classnames that FontAwesome provides, which are more descriptive. 

So we would pass through something like `content="fa-globe"`.

Reply:
Yeah, that sounds good